File: libs/power-log-parser/src/lib/handlers/player-name-handler.ts
import { innkeeperNames, bobTavernNames } from '../helper';
import { Regexes } from '../regexes';
import type { ParserState, StateType } from '../state/parser-state';

export class PlayerNameHandler {
	static Handle(timestamp: string, data: string, state: ParserState, stateType: StateType): boolean {
		const match = Regexes.PlayerNameAssignment.exec(data);
		if (match) {
			const playerId = parseInt(match[1], 10);
			const playerName = match[2];
			try {
				const matchingPlayer = state.getPlayers().find((player) => player.PlayerId === playerId);
				if (!matchingPlayer) {
					return false;
				}
				matchingPlayer.Name = playerName;
				matchingPlayer.InitialName = innkeeperNames.includes(playerName)
					? innkeeperNames[0]
					: bobTavernNames.includes(playerName)
						? bobTavernNames[0]
						: playerName;
				if (state.LocalPlayer?.PlayerId === playerId) {
					state.LocalPlayer.Name = playerName;
				}
				if (state.OpponentPlayer?.PlayerId === playerId) {
					state.OpponentPlayer.Name = playerName;
				}
				state.TryAssignLocalPlayer(timestamp, data);
				console.debug('Tried to assign player name', data);
			} catch (e: any) {
				console.debug('Exception while assigning player name', data);
				return false;
			}
			return true;
		}
		return false;
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Parser/Handlers/DataHandler/PlayerNameHandler.cs
using HearthstoneReplays.Enums;
using HearthstoneReplays.Parser.ReplayData.Entities;
using HearthstoneReplays.Parser.ReplayData.GameActions;
using HearthstoneReplays.Parser.ReplayData;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Action = HearthstoneReplays.Parser.ReplayData.GameActions.Action;

namespace HearthstoneReplays.Parser.Handlers
{
    internal class PlayerNameHandler
    {
        public static bool HandlePlayerName(DateTime timestamp, string data, ParserState state, StateType stateType)
        {
            var match = Regexes.PlayerNameAssignment.Match(data);
            if (match.Success)
            {
                var playerId = int.Parse(match.Groups[1].Value);
                var playerName = match.Groups[2].Value;
                try
                {
                    var matchingPlayer = state.getPlayers()
                        .Where(player => player.PlayerId == playerId)
                        .First();
                    matchingPlayer.Name = playerName;
                    matchingPlayer.InitialName = Helper.innkeeperNames.Contains(playerName)
                        ? Helper.innkeeperNames[0]
                        : Helper.bobTavernNames.Contains(playerName)
                        ? Helper.bobTavernNames[0]
                        : playerName;
                    state.TryAssignLocalPlayer(timestamp, data);
                    Logger.Log("Tried to assign player name", data);
                }
                catch (Exception e)
                {
                    Logger.Log("Exceptionw while assigning player name", data);
                    return false;
                }
                return true;
            }
            return false;
        }
    }
}
